readme.md

```diff
@@ -1,48 +1,408 @@
+          src: 'https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_120x44dp.png',
+          href: 'https://www.google.com',
+          x: 10,
+          y: 400,
+          cx: 50,
+        });
+
+        // Images can be added inline as a base64 encoded string.
+        slide.addImage(image => {
+          image
+            .data('iVBORw0KGgoA[...]Jggg')
+            .x(350)
+            .y(200);
+        });
+    });
+});
+```
+
+### Media Objects
+The pptx spec calls for support of media objects (video & audio) however presently `node-pptx` doesn't support these objects.
+
+### Shapes
+For a full list of the supported shapes check the
+[shape-types](https://github.com/heavysixer/node-pptx/blob/master/lib/shape-types.js) file.
+
+```javascript
+const PPTX = require('node-pptx');
+let pptx = new PPTX.Composer();
+
+await pptx.compose(async pres => {
+    await pres.addSlide(slide => {
+
+        //Creating a shape using the DSL.
+        slide.addShape(shape => {
+            shape
+                .type(PPTX.ShapeTypes.TRIANGLE)
+                .x(50)
+                .y(50)
+                .cx(50)
+                .cy(50);
+        });
+
+        //Creating a shape using the object-only syntax.
+        slide.addShape({ type: PPTX.ShapeTypes.TRIANGLE, x: 150, y: 50, cx: 50, cy: 50, color: '00FF00' });
+
+        // Adding a hyperlink to the shape.
+        slide.addShape({ type: PPTX.ShapeTypes.UP_ARROW, x: 500, y: 140, cx: 100, cy: 50, color: '0000FF', href: 'www.google.com' });
+    });
+});
+
+await pptx.save(`./shapes-test.pptx`);
+```
+
+### Text Boxes
+As the name suggests text can be added to the slide using `addText`.  The text box element also supports the creation of external links (which open a web browser) and internal linking (which link to another slide in the same presentation).
+
+```javascript
+const PPTX = require('node-pptx');
+let pptx = new PPTX.Composer();
+
+await pptx.compose(async pres => {
+    await pres.addSlide(slide => {
+
+        // declarative way of adding an object
+        slide.addText(text => {
+            text
+                .value('Hello World!')
+                .x(100)
+                .y(50)
+                .fontFace('Alien Encounters')
+                .fontSize(20)
+                .textColor('CC0000')
+                .textWrap('none')
+                .textAlign('left')
+                .textVerticalAlign('center')
+                .line({ color: '0000FF', dashType: 'dash', width: 1.0 })
+                .margin(0);
+        });
+
+        // plain "config" method of adding an object
+        slide.addText({ value: 'Link to google.com', x: 200, y: 300, href: 'http://www.google.com' });
+    });
+});
+
+await pptx.save(`./text-box-new-simple.pptx`);
+```
+
+To create an external link specify the full URI path as the value for the `url` key.
+```javascript
+defaultSlide.addText({ value: 'This is a hyperlink!', x: 0, y: 25, cx: 400, url: 'http://www.google.com' });
```

Comment:
"url" was changed to "href" to avoid confusion about image download url's vs. url links.

Reply:
👍